enhancements/oc/oc-login-external-oidc-integration.md

```diff
@@ -0,0 +1,238 @@
+      installHint: Please be sure that oc is defined in $PATH to be executed as credentials
+        exec plugin
+      interactiveMode: IfAvailable
+      provideClusterInfo: false
+```
+
+Moreover, `oc login` command triggers authentication process by sending a request to Project API's whoami endpoint automatically so that
+client-go talks to `oc get-token` command to obtain id token (and also refresh token, if provider supports it).
+
+There are currently 7 authentication flows grouped around public and confidential clients. Supportability (in terms of authentication flows) matrix of this feature is outlined below; 
+
+|           | Authorization Code<br/>(public)                                            | Authorization Code<br/>(confidential)                                                                    | Authorization Code + PKCE<br/>(public)                                     | Authorization Code + PKCE<br/>(confidential)                                                             | Implicit Flow<br/>(public) | Client Credentials<br/>(confidential)                                    | Refresh Token<br/>(public/confidential) | Device Code<br/>(public) | Password Grant<br/>(confidential) |
+|-----------|----------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------|----------------------------|--------------------------------------------------------------------------|-----------------------------------------|--------------------------|-----------------------------------|
+| Supported | Yes                                                                        | Yes                                                                                                      | Yes                                                                        | Yes                                                                                                      | No                         | No                                                                       | Yes                                     | No                       | No                                |
+| Comment   |                                                                            | Only if provider supports this[1]                                                                        |                                                                            | Only if provider supports this[1]                                                                        | Deprecated[3]              | Mostly used in CI systems that <br/>client-secret can be stored securely | If provider supports it                 |                          | Deprecated[4]                     |
```

Comment:
let's use this column to break down each cell with 

1. keycloak - [default|not-default], [configurable|not-configurable]
2. entra -  [default|not-default], [configurable|not-configurable]